datalad_dataverse/remote.py

```diff
@@ -92,111 +136,311 @@ def api(self):
 
         return self._api
 
+    @property
+    def dataset(self):
+        """Full JSON record of the dataverse dataset.
+
+        This is not (yet) a cache! It only holds the first response.
+        For now the only thing we work with is the resulting file list,
+        which is initialized from this and then updated upon upload/removal of
+        files.
+
+        Todo
+        ----
+        This may or may not change when the remote is feature complete. If it
+        turns out to stay that way, we should likely fuse this with
+        `self.files_all`/`self.files_latest` and not keep all of it in memory.
+        """
+
+        if self._dataset is None:
+            # This delivers a full record of all known versions of this dataset.
```

Comment:
Does that mean this is expected to be a huge record?

What is the tradeoff re request latency?

If I have a dataset with 10k files, and a second version changes the README, do I have to download JSON records for 20k files?